### PR TITLE
Add task_id field to ExecutionParameters

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Adding a `task_id` field to the `ExecutionParameters` object to read more of the environment variables set by the [backend](https://github.com/flyteorg/flyteplugins/blob/60b94c688ef2b98aa53a9224b529ac672af04540/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go).

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2490

### CODE REVIEW

**Comment by wild-endeavor** [comment]
@dylanwilder take a look?

File: tests/flytekit/unit/core/test_context_manager.py [hunk 1]
```diff
@@ -11,7 +12,9 @@
     SecretsConfig,
     SerializationSettings,
 )
-from flytekit.core.context_manager import FlyteContext, FlyteContextManager, SecretsManager
+from flytekit.core import mock_stats, utils
```
[the remark this answers is not in the document]
Comment (wild-endeavor) [reply tests/flytekit/unit/core/test_context_manager.py]: thanks

**Review by pingsutw — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
